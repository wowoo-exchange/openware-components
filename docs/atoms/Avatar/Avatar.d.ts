import * as React from 'react';
interface AvatarProps {
    /**
     * Property for an alt of the image. If source is defined, it will be hidden
     */
    title: string;
    /**
     * Additional class name. By default element receives `cr-avatar` class
     */
    className?: string;
    /**
     * Property for an image. If it is not undefined, image will be hidden and there will be displayed alt instead of image
     */
    source?: string;
}
/**
 * Component to display cryptobase-react Avatar
 */
declare const Avatar: React.FunctionComponent<AvatarProps>;
export { Avatar, AvatarProps, };
