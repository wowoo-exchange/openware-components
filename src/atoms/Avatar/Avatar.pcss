.cr-avatar {
  width: var(--avatar-component-width);

  img {
    width: auto;
  }

  &__image {
    border: 1px solid var(--avatar-border-alt-color);
    height: var(--avatar-alt-height);
    width: var(--avatar-alt-width);
    background-color: var(--avatar-background-alt-color);
    text-align: center;
    vertical-align: middle;
    justify-content: center;
    max-width: 2ch;
    padding: var(--gap);
    font-size: var(--avatar-alt-font-size);
    line-height: normal;
    color: var(--avatar-alt-font-color);
    font-weight: normal;
  }
}
