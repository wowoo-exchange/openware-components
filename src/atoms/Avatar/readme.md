Example of usage:

```js
  <Avatar title="SS" />
```
