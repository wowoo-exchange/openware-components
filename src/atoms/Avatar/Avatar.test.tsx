import { shallow } from 'enzyme';
import * as React from 'react';
import { Avatar, AvatarProps } from './Avatar';

const defaultProps: AvatarProps = {
  source: undefined,
  title: 'KE',
};

const setup = (props: Partial<AvatarProps> = {}) =>
  shallow(<Avatar {...{ ...defaultProps, ...props }} />);

describe('Avatar', () => {
  it('should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should have correct className', () => {
    const wrapper = setup();
    expect(wrapper.hasClass('cr-avatar')).toBeTruthy();
  });

  it('should pass along supplied className', () => {
    const className = 'some-class';
    const wrapper = setup({ className });
    expect(wrapper.hasClass(className)).toBeTruthy();
  });

  it('should render correct component depends on props', () => {
    let wrapper = setup();
    let images = wrapper.find('img');
    expect(images.length).toBe(0);
    wrapper = setup({source: 'source'});
    images = wrapper.find('img');
    expect(images.length).toBe(1);
  });
});
