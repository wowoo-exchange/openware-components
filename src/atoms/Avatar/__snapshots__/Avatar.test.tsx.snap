// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render 1`] = `
<div
  className="cr-avatar"
>
  <div
    className="cr-avatar__image"
  >
    KE
  </div>
</div>
`;
